src/Bicep.Core/TypeSystem/OperationReturnTypeEvaluator.cs

```diff
@@ -282,23 +282,15 @@ protected NumericComparisonEvaluator(BinaryOperator @operator, string armFunctio
     {
         IntegerLiteralType literal => literal.Value,
         IntegerType @int => @int.MinValue,
-        UnionType union => union.Members.Select(m => MinValue(m.Type)) switch
-        {
-            IEnumerable<long?> minOfEachMember when minOfEachMember.All(m => m.HasValue) => minOfEachMember.Min(),
-            _ => null,
-        },
```

Comment:
I found out that applying `Max` or `Min` to an empty collection with nullable item types doesn't result in an exception; instead, it returns null. If the collection is not empty, the null items are get ignored automatically. This makes the use of a switch expression redundant.